virtcontainers/network.go

```diff
@@ -247,21 +255,21 @@ func (endpoint *VirtualEndpoint) Attach(h hypervisor) error {
 func (endpoint *VirtualEndpoint) Detach(netNsCreated bool, netNsPath string) error {
 	// The network namespace would have been deleted at this point
 	// if it has not been created by virtcontainers.
-	if !netNsCreated {
+	if !netNsCreated && netNsPath != "" {
```

Comment:
What's this modification for？

Reply:
This is because when use `disable_new_netns`, netNsCreated is false since we don't create a netns. Without this modification, `Detach` function returns without cleaning device. The tap device will leave in the host netns.